pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -35,32 +35,49 @@ import (
 
 	appsv1alpha1 "github.com/karmada-io/karmada/pkg/apis/apps/v1alpha1"
 	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util"
 	"github.com/karmada-io/karmada/pkg/util/names"
 )
 
 const (
 	// ControllerName is the controller name that will be used when reporting events.
 	ControllerName = "workload-rebalancer"
+
+	ttlAfterFinishedWorkerNum = 1
 )
 
 // RebalancerController is to handle a rebalance to workloads selected by WorkloadRebalancer object.
 type RebalancerController struct {
-	Client client.Client
+	Client             client.Client // used to operate WorkloadRebalancer resources from cache.
+	ControlPlaneClient client.Client // used to fetch arbitrary resources from api server.
+
+	ttlAfterFinishedWorker util.AsyncWorker
 }
 
 // SetupWithManager creates a controller and register to controller manager.
 func (c *RebalancerController) SetupWithManager(mgr controllerruntime.Manager) error {
 	var predicateFunc = predicate.Funcs{
-		CreateFunc: func(event.CreateEvent) bool { return true },
+		CreateFunc: func(e event.CreateEvent) bool {
+			c.ttlAfterFinishedWorker.Add(client.ObjectKey{Name: e.Object.GetName()})
+			return true
+		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			oldObj := e.ObjectOld.(*appsv1alpha1.WorkloadRebalancer)
 			newObj := e.ObjectNew.(*appsv1alpha1.WorkloadRebalancer)
+			c.ttlAfterFinishedWorker.Add(client.ObjectKey{Name: newObj.GetName()})
 			return !reflect.DeepEqual(oldObj.Spec, newObj.Spec)
 		},
 		DeleteFunc:  func(event.DeleteEvent) bool { return false },
 		GenericFunc: func(event.GenericEvent) bool { return false },
 	}
 
+	ttlAfterFinishedWorkerOptions := util.Options{
+		Name:          "ttl-after-finished-worker",
+		ReconcileFunc: c.deleteExpiredRebalancer,
+	}
+	c.ttlAfterFinishedWorker = util.NewAsyncWorker(ttlAfterFinishedWorkerOptions)
+	c.ttlAfterFinishedWorker.Run(ttlAfterFinishedWorkerNum, context.Background().Done())
```

Comment:
I am still not sure if we should introduce another queue to check the TTL yet, just feel a little bit heavy. I will look at the Job controller from Kubernetes, and then get back on this.